frontend/src/components/ViewTrips/trip.js

```diff
@@ -63,8 +67,28 @@ const Trip = (props) => {
   const name = props.tripData.name;
   const description = props.tripData.description;
   const destination = props.tripData.destination;
-  const collaboratorEmailsStr =
-      getCollaboratorEmails(props.tripData.collaborators);
+  const collaboratorUidArr = props.tripData.collaborators;
+  const [collaboratorEmailsStr, setCollaboratorEmailsStr] = useState('');
+
+  useEffect(() => {
+    // Only set state collaboratorEmailsStr if component is mounted. This is
+    // a precautionary to mitigate warnings that occur when setting state on
+    // a component that has already unmounted. See more here
+    // https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component.
+    let componentStillMounted = true;
+
+    async function fetchCollaboratorEmails() {
+      let collaboratorEmailArr =
+          await authUtils.getUserEmailArrFromUserUidArr(collaboratorUidArr);
+      collaboratorEmailArr = moveCurUserEmailToFront(collaboratorEmailArr);
+      if (componentStillMounted) {
+        setCollaboratorEmailsStr(collaboratorEmailArr.join(', '));
+      }
+    }
+
+    fetchCollaboratorEmails();
+    return () => { componentStillMounted = false; };
```

Comment:
So useEffect is returning a lambda function that changes componentStillMounted to false? This is getting more complicated -- could you add a comment to give more context on how this used?

Reply:
So yeah this looks pretty weird but most of this is documented in React's documentation on [effect hook with cleanup](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup). So because this is a react specific capability and not a weird thing I did, should I still add a comment? If so, it is no problem, I just want to understand what the best practice is in this case.

Reply:
I think it might be good to just make a simple comment above the return statement as `// Cleanup function.` or something similar.

Reply:
Yeah I think a short comment would be good so that people unfamiliar with React-specifics can still get the gist of it.

Reply:
Yeah that makes sense. Added.